src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -364,7 +364,8 @@ private static IUnboundLambdaExpression CreateUnboundLambdaOperation(UnboundLamb
             // an IUnboundLambdaExpression, you need to look at the parent IConversionExpression.
             ITypeSymbol type = null;
             Optional<object> constantValue = ConvertToOptional(unboundLambda.ConstantValue);
-            return new UnboundLambdaExpression(isInvalid, syntax, type, constantValue);
+            Lazy<ILambdaExpression> internalLambda = new Lazy<ILambdaExpression>(() => CreateBoundLambdaOperation(unboundLambda.BindForErrorRecovery()));
```

Comment:
Fixed. We now retrieve the node from the SemanticModel, and the SemanticModel has an assertion that the bound node that it will give to the factory will never be an UnboundLambda.